### PR TITLE
Optimize resourcegroup test, make test separately and readable

### PR DESCRIPTION
**Pull request by Shoothzj**
### Motivation
I'm working on #12499, need to modify the corresponding tests. But currently, these test files are hard to read.
- Some unused method
- lowercase ClassName and typos
- Tests all in one method
I want to optimize the tests first

### Modifications
- separately the tests
- remove unused method
- rename the classname
- fix the typo

### Documentation

Need to update docs? 
  
- [x] `no-need-doc` 
  
  Just tests optimize

### CODE REVIEW

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ResourceGroup.java [hunk 1]
```diff
@@ -20,7 +20,8 @@
 
 import lombok.Data;
 
-@Data public class ResourceGroup {
+@Data
+public class ResourceGroup {
```
Comment (tomscut) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ResourceGroup.java:24]: Don't need to be changed here.

Comment (Shoothzj) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ResourceGroup.java:24]: It's more readable and same with other code's style.

Comment (tomscut) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ResourceGroup.java:24]: Sorry, I made a mistake.

**Review by tomscut — APPROVED** [review]
LGTM.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**